From 9971984d96e48931bdcabda9d79ab2985633e5dc Mon Sep 17 00:00:00 2001
From: YOSHIOKA Takuma <tashioka.256@gmail.com>
Date: Tue, 23 Jan 2018 15:46:47 +0900
Subject: [PATCH 1/3] Improve kana conversion table and add tests for
 hankaku-kana conversion

Added small-ka, small-ke, and some symbols used frequently in Japanese.
---
 libskk/util.vala | 12 +++++++++++-
 tests/basic.c    |  4 ++++
 2 files changed, 15 insertions(+), 1 deletion(-)

diff --git a/libskk/util.vala b/libskk/util.vala
index 9fd6e81..281f739 100644
--- a/libskk/util.vala
+++ b/libskk/util.vala
@@ -70,7 +70,17 @@ namespace Skk {
             {'ッ', "っ", "ｯ"},
             {'ャ', "ゃ", "ｬ"}, {'ュ', "ゅ", "ｭ"}, {'ョ', "ょ", "ｮ"},
             {'ヮ', "ゎ", null},
-            {'ヴ', "う゛", "ｳﾞ"}, {'ヵ', null, null}, {'ヶ', null, null}
+            {'ヴ', "う゛", "ｳﾞ"}, {'ヵ', "ゕ", null}, {'ヶ', "ゖ", null},
+            {'．', "．", "."},
+            {'，', "，", ","},
+            {'。', "。", "｡"},
+            {'「', "「", "｢"},
+            {'」', "」", "｣"},
+            {'、', "、", "､"},
+            {'・', "・", "･"},
+            {'ー', "ー", "ｰ"},
+            {'゛', "゛", "ﾞ"},
+            {'゜', "゜", "ﾟ"}
         };
 
         const KanaTableEntry[] HankakuKatakanaSubstitute = {

From 00a54efa82c4588c15151bb51cc535d1c0f6b1ec Mon Sep 17 00:00:00 2001
From: YOSHIOKA Takuma <tashioka.256@gmail.com>
Date: Tue, 23 Jan 2018 18:05:30 +0900
Subject: [PATCH 2/3] Auto-start henkan by hankaku katakana "wo", comma, and
 period

---
 libskk/state.vala | 3 ++-
 tests/basic.c     | 2 ++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/libskk/state.vala b/libskk/state.vala
index 0a72f2c..d094547 100644
--- a/libskk/state.vala
+++ b/libskk/state.vala
@@ -20,7 +20,8 @@ using Gee;
 
 namespace Skk {
     const string[] AUTO_START_HENKAN_KEYWORDS = {
-        "を", "、", "。", "．", "，", "？", "」",
+        "を", "ヲ", "、", "。", "．", "，", "？", "」",
+        "ｦ", "､", "｡", "｣",
         "！", "；", "：", ")", ";", ":", "）",
         "”", "】", "』", "》", "〉", "｝", "］",
         "〕", "}", "]", "?", ".", ",", "!"

From 2d3ae61210ec0b5c0fceed1289d994b18ed4709f Mon Sep 17 00:00:00 2001
From: YOSHIOKA Takuma <tashioka.256@gmail.com>
Date: Tue, 23 Jan 2018 18:16:58 +0900
Subject: [PATCH 3/3] Let punctuations half-width in hankaku-katakana mode

---
 libskk/rom-kana.vala | 5 ++++-
 tests/basic.c        | 4 ++++
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/libskk/rom-kana.vala b/libskk/rom-kana.vala
index 035467e..c358f05 100644
--- a/libskk/rom-kana.vala
+++ b/libskk/rom-kana.vala
@@ -254,7 +254,10 @@ namespace Skk {
                 if (index >= 0) {
                     index = PERIOD_RULE[period_style].index_of_nth_char (index);
                     unichar period = PERIOD_RULE[period_style].get_char (index);
-                    _output.append_unichar (period);
+                    string kana_period = Util.convert_by_input_mode(
+                        period.to_string(),
+                        (InputMode)kana_mode);
+                    _output.append (kana_period);
                     _preedit.erase ();
                     current_node = rule.root_node;
                     return true;
